error/s2n_errno.c

```diff
@@ -14,155 +14,166 @@
  */
 
 #include <strings.h>
+#include <stdio.h>
 #include <stdlib.h>
-
+#include <string.h>
 #include "error/s2n_errno.h"
 
 #include <s2n.h>
+#define __USE_GNU
+#include <search.h>
 
 __thread int s2n_errno;
 __thread const char *s2n_debug_str;
 
 static const char *no_such_language = "Language is not supported for error translation";
 static const char *no_such_error = "Internal s2n error";
 
-struct s2n_error_translation {
+typedef struct _s2n_error_translation {
     int errno_value;
+    const char *error_name;
     const char *str;
-};
+}s2n_error_translation;
+
+#define KEY_LEN 10
+#define KEY_BASE 16
+#define ERROR_STRING(x) x, #x
```

Comment:
Would it be neater to have the macro do everything we need? e.g.

ERROR_STRING(x, y)  { (x) , (#x), (y) },

Reply:
> Would it be neater to have the macro do everything we need? e.g.
> 
> ERROR_STRING(x, y) { (x) , (#x), (y) },

Good idea.